src/cargo/core/compiler/mod.rs

```diff
@@ -894,10 +894,20 @@ fn build_base_args(
     cmd.args(&lto_args(cx, unit));
 
     // This is generally just an optimization on build time so if we don't pass
-    // it then it's ok. As of the time of this writing it's a very new flag, so
-    // we need to dynamically check if it's available.
-    if cx.bcx.target_data.info(unit.kind).supports_split_debuginfo {
-        if let Some(split) = split_debuginfo {
+    // it then it's ok. The values for the flag (off, packed, unpacked) may be supported
+    // or not depending on the platform, so availability is checked per-value.
+    // For example, at the time of writing this code, on Windows the only stable valid
+    // value for split-debuginfo is "packed", while on Linux "unpacked" is also stable.
+    if let Some(split) = split_debuginfo {
+        if cx
+            .bcx
+            .target_data
+            .info(unit.kind)
+            .support_split_debuginfo
+            .iter()
+            .find(|sup| sup.as_str() == split.as_str())
+            .is_some()
```

Comment:
(nit) Could we extract this piece of code and comment into a method under `TargetInfo`? So that we could turn field `support_split_debuginfo` into a non-`pub` one.